k2/csrc/ragged_utils.cu

```diff
@@ -156,7 +156,7 @@ RaggedShape IntersperseRaggedLayer(int32_t layer,
              row_splits_ptrs_data[src][pos];
         row_splits_data[i] = this_size;
       };
-      EvalDevice(c, new_num_rows + 1, lambda_get_sizes);
+      EvalDevice(c, new_num_rows, lambda_get_sizes);
```

Comment:
Line 155
```
        int32_t this_size = row_splits_ptrs_data[src][pos + 1] -
             row_splits_ptrs_data[src][pos];
```

`row_splits_ptrs_data[src][pos + 1]` is problematic when i equals to `new_num_rows`.

This PR fixes that.